app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/map/BeaconMapBackground.java

```diff
@@ -0,0 +1,214 @@
+import com.nexenio.bleindoorpositioning.location.projection.CanvasProjection;
+
+public class BeaconMapBackground {
+
+    private Bitmap imageBitmap;
+
+    private double metersPerPixel;
+
+    private double bearing;
+
+    private Location topLeftLocation;
+    private Location bottomRightLocation;
+
+    private Point topLeftPoint;
+    private Point bottomRightPoint;
+
+    private CanvasProjection projection = new CanvasProjection();
+
+    private BeaconMapBackground(Bitmap imageBitmap) {
+        this.imageBitmap = imageBitmap;
+        projection.setCanvasWidth(imageBitmap.getWidth());
+        projection.setCanvasHeight(imageBitmap.getHeight());
+        topLeftPoint = new Point(0, 0);
+        bottomRightPoint = new Point(imageBitmap.getWidth(), imageBitmap.getHeight());
+    }
+
+    public Location getLocation(Point point) {
+        return getLocation(point, topLeftLocation, topLeftPoint, metersPerPixel, bearing);
+    }
+
+    public Point getPoint(Location location) {
+        return getPoint(location, topLeftLocation, topLeftPoint, metersPerPixel, bearing);
+    }
+
+    public static float getMetersPerPixel(Location firstReferenceLocation, Point firstReferencePoint, Location secondReferenceLocation, Point secondReferencePoint) {
+        double distanceInPixels = getPixelDistance(firstReferencePoint, secondReferencePoint);
+        double distanceInMeters = firstReferenceLocation.getDistanceTo(secondReferenceLocation);
+        return (float) (distanceInMeters / distanceInPixels);
+    }
+
+    public static double getBearing(Location firstReferenceLocation, Point firstReferencePoint, Location secondReferenceLocation, Point secondReferencePoint) {
+        double locationBearing = getBearing(firstReferenceLocation, secondReferenceLocation);
+        double pointBearing = getBearing(firstReferencePoint, secondReferencePoint);
+        return ((locationBearing - pointBearing) + 360) % 360;
+    }
+
+    public static double getBearing(Point firstPoint, Point secondPoint) {
+        double angle = Math.atan2(secondPoint.y - firstPoint.y, secondPoint.x - firstPoint.x) * 180 / Math.PI;
+        return (angle + 90) % 360;
+    }
+
+    public static double getBearing(Location firstLocation, Location secondLocation) {
+        return firstLocation.getAngleTo(secondLocation);
+    }
+
+    public static double getPixelDistance(Point firstPoint, Point secondPoint) {
+        int deltaX = Math.abs(secondPoint.x - firstPoint.x);
+        int deltaY = Math.abs(secondPoint.y - firstPoint.y);
+        return Math.sqrt((deltaX * deltaX) + (deltaY * deltaY));
+    }
+
+    public static Location getLocation(Point point, Location referenceLocation, Point referencePoint, double metersPerPixel, double backgroundBearing) {
+        double distance = metersPerPixel * getPixelDistance(referencePoint, point);
+        double bearing = (getBearing(referencePoint, point) + backgroundBearing + 360) % 360;
```

Comment:
As bearings can not be negative (at least they should the way we use them) the `+360` is unnecessary. (positive number + positive number will stay positive).